plugins/utils/competitest.nix

```diff
@@ -0,0 +1,202 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "competitest";
+  originalName = "competitest.nvim";
+  defaultPackage = pkgs.vimPlugins.competitest-nvim;
+
+  maintainers = [helpers.maintainers.svl];
+
+  extraOptions = with lib.types; {
+    extraConfig = helpers.mkNullOrOption (attrsOf anything) ''
+      Extra config options for competitest.nvim
+    '';
+  };
+
```

Comment:
This is not needed as the `settings` option will be self-extendable.
The user will be able to set 
`plugins.competitest.settings.foo-bar = 42;`
even if you haven't declared this `foo-bar` option below.

Learn more [here](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md).